src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -428,6 +467,100 @@ void AccessControlAttribute::OnEntryChanged(const SubjectDescriptor * subjectDes
     ChipLogError(DataManagement, "AccessControlCluster: event failed %" CHIP_ERROR_FORMAT, err.Format());
 }
 
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+CHIP_ERROR AccessControlAttribute::ReadCommissioningArl(AttributeValueEncoder & aEncoder)
+{
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+        auto entries = accessRestrictionProvider->GetCommissioningEntries();
+
+        for (auto & entry : entries)
+        {
+            ArlStorage::EncodableEntry encodableEntry(entry);
+            ReturnErrorOnFailure(encoder.Encode(encodableEntry));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR AccessControlAttribute::ReadArl(AttributeValueEncoder & aEncoder)
+{
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+        for (auto & info : Server::GetInstance().GetFabricTable())
+        {
+            auto fabric = info.GetFabricIndex();
+            // get entries for fabric
+            const std::vector<AccessRestrictionProvider::Entry> * entries = nullptr;
+            ReturnErrorOnFailure(accessRestrictionProvider->GetEntries(fabric, entries));
+            VerifyOrReturnError(entries != nullptr, CHIP_ERROR_INCORRECT_STATE);
+            for (auto & entry : *entries)
+            {
+                ArlStorage::EncodableEntry encodableEntry(entry);
+                ReturnErrorOnFailure(encoder.Encode(encodableEntry));
+            }
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+void AccessControlAttribute::CommissioningRestrictionListChanged()
+{
+    MatterReportingAttributeChangeCallback(0, AccessControlCluster::Id, AccessControlCluster::Attributes::CommissioningARL::Id);
+}
+
+void AccessControlAttribute::RestrictionListChanged(FabricIndex fabricIndex)
+{
+    CHIP_ERROR err;
+
+    MatterReportingAttributeChangeCallback(0, AccessControlCluster::Id, AccessControlCluster::Attributes::Arl::Id);
+
+    ArlChangedEvent event{ .fabricIndex = fabricIndex };
```

Comment:
So spec says these events are generated "whenever its ARL attribute data is changed by the device."

It doesn't define the fabric-association of the events, so this code is just making up some non-spec-defined behavior.  Maybe that behavior is the right one; I can't tell.  Spec needs to define this.

Reply:
This event has now been removed from the spec so I will remove it from the code shortly.